packages/circuit-ui/components/PhoneNumberInput/PhoneNumberInput.tsx

```diff
@@ -240,8 +240,21 @@ export const PhoneNumberInput = forwardRef<
       ) {
         return;
       }
+
+      const selectedCountry = countryCodeRef?.current?.value;
+      if (!selectedCountry) {
+        return;
+      }
+      const code = countryCode.options
+        // Match longer, more specific country codes first
+        .sort((a, b) => a.code.length - b.code.length)
+        .find(({ country }) => country === selectedCountry)?.code;
```

Comment:
The lookup uses the country ISO code, so sorting by the country code phone prefix is unnecessary.

```suggestion
      const code = countryCode.options
        .find(({ country }) => country === selectedCountry)?.code;
```